src/init.cpp

```diff
@@ -650,6 +650,7 @@ void SetupServerArgs(ArgsManager& argsman, bool can_listen_ipc)
 
 
     argsman.AddArg("-blockmaxweight=<n>", strprintf("Set maximum BIP141 block weight (default: %d).", MAX_BLOCK_WEIGHT), ArgsManager::ALLOW_ANY, OptionsCategory::BLOCK_CREATION);
+    argsman.AddArg("-maxcoinbaseweight=<n>", strprintf("Set the block maximum reserved coinbase transaction weight (default: %d).", node::BlockCreateOptions{}.coinbase_max_additional_weight), ArgsManager::ALLOW_ANY, OptionsCategory::BLOCK_CREATION);
```

Comment:
a45af6af4e7a59cd14ef6e16b8e5e180fa9d564d

Question about this phrasing: what is "maximum" about it? If a user sets `-maxcoinbaseweight=1000` then 1000 weight will always be reserved in block templates, and if the actual coinbase ends up being 50 weight instead of 1000 that extra weight will just be unused and the final block will be smaller than maximum size.

Reply:
Correct.

Reply:
Perhaps `-coinbaseweightreservation` is a better name.

Reply:
I've taken your suggestion @Sjors.